src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -449,8 +450,12 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // ptr and offset_ptr must be in bounds of the same allocated object. This means all of the
         // memory between these pointers must be accessible. Note that we do not require the
         // pointers to be properly aligned (unlike a read/write operation).
-        let min_ptr = if offset_bytes >= 0 { ptr } else { offset_ptr };
-        let size = offset_bytes.checked_abs().unwrap();
+        let (min_ptr, size) = if offset_bytes >= 0 {
+            (ptr, offset_bytes as u64)
```

Comment:
Please avoid `as`, it might lose information.

Reply:
Fixed to use `u64::try_from(offset_bytes).unwrap()`